src/DefaultBuilder/src/StaticWebAssetsLoader.cs

```
@@ -0,0 +1,96 @@
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.FileProviders;
+
+namespace Microsoft.AspNetCore
+{
+    internal class StaticWebAssetsLoader
+    {
+        internal const string StaticWebAssetsManifestName = "Microsoft.AspNetCore.StaticWebAssets.xml";
+
+        internal static void UseStaticWebAssets(IWebHostEnvironment environment)
+        {
+            using (var manifest = ResolveManifest(environment))
+            {
+                if (manifest != null)
+                {
+                    UseStaticWebAssetsCore(environment, manifest);
+                }
+            }
+        }
+
+        internal static void UseStaticWebAssetsCore(IWebHostEnvironment environment, Stream manifest)
+        {
+            var staticWebAssetsFileProvider = new List<IFileProvider>();
+            var webRootFileProvider = environment.WebRootFileProvider;
+
+            var additionalFiles = StaticWebAssetsReader.Parse(manifest)
+                .Select(cr => new StaticWebAssetsFileProvider(cr.BasePath, cr.Path))
+                .OfType<IFileProvider>() // Upcast so we can insert on the resulting list.
+                .ToList();
+
+            if (additionalFiles.Count == 0)
+            {
+                return;
+            }
+            else
+            {
+                additionalFiles.Insert(0, webRootFileProvider);
+                environment.WebRootFileProvider = new CompositeFileProvider(additionalFiles);
+            }
+        }
+
+        internal static Stream ResolveManifest(IWebHostEnvironment environment)
+        {
+            // We plan to remove the embedded file resolution code path in
+            // a future preview.
+            Assembly assembly = null;
+            try
+            {
+                assembly = Assembly.Load(environment.ApplicationName);
```

Why swallow the failure? Can we not?

---

As mentioned in the comment, this code-path is going away in preview7.